### DISCUSSION TITLE
First official alpha release

### DISCUSSION THREAD

**Discussion by wormstein** [EdNekebno/DGTCentaurMods#80] [General]
Hello,
please push all changes you plan to have in the alpha release by tomorrow afternoon. I will tag a snapshot of the master branch.
Also let me know if there is need for any changes in the installation process that you noticed while testing sp I could update the bui;d process.

This version will install only and wont support removing and reinstalling without a full reimagine of RPi. Upgrade of it might be possibe in the future.

Cheers

**Comment by EdNekebno** [comment]
Oooh exciting...

Sounds good. I'm a bit under the weather today, but I'll tidy up my db/centaur.ini and push either a bit later today or in the morning. Then maybe I won't give my lichess api key away again :) There are some important bug fixes amongst the additional features.

I didn't notice anything previously, but haven't tried in the board. I notice that raspbian has been updated recently too. Then I think we just accept that some manual tweaks might be necessary (it's an alpha. we can always fix/enhance things next time).

**Comment by EdNekebno** [comment]
Oh...the only one thing I can think of that is different is that engines upload to the "engines" folder in DGTCentaurMods (not the /home/pi/centaur/engines one) - it might be worth ensuring it is writable for whatever user flask is running as.

**Comment by wormstein** [comment]
The release it's there. All checks tell me it sould work whn inside the board. I couldn't do that test yet.

**Reply by EdNekebno** [reply]
Awesome work. That's a great leap for the project. Not even a hospital visit stops you! It's good that we have an official, easier way to install. It's getting late so I'll try to give it a go tomorrow, although it promises to be a busy day for me, so if not saturday.

**Reply by wormstein** [reply]
Our version of SF is integrated, btw.

**Comment by EdNekebno** [comment]
Unfortunately most things are broken when you install it on latest raspbian and put it in the board :( I've made the following changes on my sdcard to make it work (I think....mostly):

DGTCentaurMods.service - won't start as python3.9 is in bullseye, not 3.7 - modified python3.7 command line in the file to python3
Nothing could write to the fen.log and would error out - chown pi /home/pi/centaur; chgrp pi /home/pi/centaur
For bluetooth:
   added PyBluez==0.23.0   and   psutil==5.8.0  to requirements.txt and pip3 install -r requirements.txt
   apt install libbluetooth-dev
   apt install bluez-tools
   Modified bluetooth.service:
     ExecStart=/usr/libexec/bluetooth/bluetoothd --compat --noplugin=sap,battery
     ExecStartPost=/usr/bin/sdptool add SP
   Also in dbus-org.bluez.service:
     ExecStart=/usr/libexec/bluetooth/bluetoothd --compat --noplugin=sap,battery
     ExecStartPost=/usr/bin/sdptool add SP
   (I suspect modifying the bluetooth.service is unnecessary, it seemed to be the second one that worked but I just did both. It's possible this was installed by the apt install commands)
   In /etc/bluetooth/main.conf
   Uncommented:
     DiscoverableTimeout = 0
     PairableTimeout = 0
   Added:
     JustWorksRepairing=always

For problems finding the python module when I was testing I did this again: cd /usr/lib/python3/dist-packages; ln -s /home/pi/DGTCentaurMods DGTCentaurMods

If I can find somewhere to put it, I can probably also take my wifi key out and image this sd-card.

**Reply by patchamama** [reply]
Thank you for the tips... Maybe "Wormstein" can fix it in the deb creator process...

**Reply by wormstein** [reply]
Good feedback. I actually tested on old Raspbian... Having a fresh SD card around i just used that.

I'll go over each bug and see what I needs adjusting.

**Reply by wormstein** [reply]
I have created an issue with your feedback.

**Comment by wormstein** [comment]
My RPi 2 W has no GPIO pins yet and I'm not very mobile yet to go to a skilled guy... So I'll try to cover all the above without testing inside the board.

**Reply by EdNekebno** [reply]
No problem. I can always test it. I have written in the board functions findings thread though that I have toasted my board at the moment :(

**Comment by wormstein** [comment]
Sorry for the delay. They just want me in the clinic everyday to check dressings, wound, etc... It's crazy.

**Reply by EdNekebno** [reply]
Don't be sorry. Health first...always.Hope you're gradually recovering.

**Reply by wormstein** [reply]
Hate to tell you this. After the clinic today and a CT scan they decided to hold me in for few days to avoid some complications...

Anyway, great job fixing the address issue. I was thinking when I'm back home to add a proper license header on every file in the project and force people follow the license. There is too much effort you put in that gets in someone else's project...

**Reply by EdNekebno** [reply]
:( Rest up.
Yeh, the headers is sensible I think. Not that I'm sure it'll stop people just taking and stripping it. I'm happy for people to use it following the license. But there's a particular project I guess you know about, with a rather embellished "history". Honestly what concerns me most is that code is presented as something it is not to end users. Then things that we are clear are early code here get into something as if it's a "final release" that someone else wrote. Some time somebody is going to get a nasty shock and we might not be able to help. Luckily there are minimal users of that and as it's a fairly specific group and I have a plan where we can fix it for users in the future. ;)

**Reply by EdNekebno** [reply]
For info...I got annoyed with the millenium eboard emulation I wrote last time I updated being renamed "Bearlink" and presented as if it's their own work and an almost final thing. I've updated the README with a note on these things. I want people to be clear they're not installing production software, they're installing versions of our alpha code. I will upload the board address code soon, so I'll do the headers into the files too. As the board address code affects so many files, it leaves no choice but to include the headers :)

**Reply by wormstein** [reply]
I feel your frustration!! As i am disabled for now I can only watch what's going on and see this too. I burn from inside out.
I like the changes you made to readme and files. I feel sorry cannot help more for now BUT these changes are just the begining.

We try very hard not to break rules with DGT company for example so they should follow license at least from now on. There are hours of hard work and testing in this now.

And i am not afraid to say it here (i know they read this) i might raise a license infringement with GitHub!! Well allow time to comply though...
